### PR TITLE
Add sparse inference and training benchmarks with sparse kernel

### PR DESCRIPTION
**Pull request by kalyc**
### Summary
Re-Added sparse benchmark results

### PR Overview

- [ ] This PR requires new unit tests [y/n] (make sure tests are included)
- [ ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [ ] This PR is backwards compatible [y/n]
- [ ] This PR changes the current API [y/n]

`Note - Previous PR also contained scripts and was resulting in git conflicts during rebase so opening another PR`

### CODE REVIEW

File: benchmark/sparse/linear_regression/inference_benchmark.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+| P3.8XLarge |   1  | 128 | 8 | 7
+| P3.8XLarge |   1  | 256 | 8 | 7
+| P3.8XLarge |   1  | 512 | 8 | 7
+| P3.8XLarge |   1  | 1024 | 7 | 8
+
+##### Memory consumed
+| Instance Type | GPUs  | Batch Size | Keras-MXNet (GPU Memory (MB)) | Keras-TensorFlow (GPU Memory (MB))  |
+|-----|-----|-----|-----|-----|
+| P3.8XLarge |   1  | 64  | 966.7 | 3322.9
+| P3.8XLarge |   1  | 128 | 970.9 | 3053.4
```
Comment (roywei) [inline benchmark/sparse/linear_regression/inference_benchmark.md]: I observe TF backend always take all available GPU memories by default. around 15388MiB for each GPU. This is expected no matter what batch size you use. Please double check, especially double check `nvidia-smi` is used during the middle of inference and training. You will get different numbers during data prep and evaluation (model.evaluate(xxx))

File: benchmark/sparse/linear_regression/training_benchmark.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+| P3.8XLarge |   1  | 1024 | 8 | 8
+
+##### Memory consumed
+| Instance Type | GPUs  | Batch Size | Keras-MXNet (GPU Memory (MB)) | Keras-TensorFlow (GPU Memory (MB))  |
+|-----|-----|-----|-----|-----|
+| P3.8XLarge |   1  | 64  | 966.7 | 2545.9
+| P3.8XLarge |   1  | 128 | 966.7 | 2323.6
+| P3.8XLarge |   1  | 256 | 970.9 | 3566.2
+| P3.8XLarge |   1  | 512 | 999.6 | 3521.1
+| P3.8XLarge |   1  | 1024 | 991.9 | 3566.2
```
Comment (roywei) [inline benchmark/sparse/linear_regression/training_benchmark.md]: same above, I get 15388MiB memory per GPU. TF is taking all mem in all GPU by default.

File: benchmark/sparse/linear_regression/inference_benchmark.md (hunk 1 above)
Comment (kalyc) [reply benchmark/sparse/linear_regression/inference_benchmark.md]: Yes updated the numbers

File: benchmark/sparse/linear_regression/training_benchmark.md (hunk 2 above)
Comment (kalyc) [reply benchmark/sparse/linear_regression/training_benchmark.md]: Yes updated the numbers

File: benchmark/sparse/linear_regression/inference_benchmark.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+##### Speed
+| Instance Type | GPUs  | Batch Size  | Keras-MXNet (Time/Batch) | Keras-TensorFlow Time/Batch)  |
+|-----|-----|-----|-----|-----|
+| P3.8XLarge |   1  | 64  | 3.0 sec | 2.0 sec
+| P3.8XLarge |   1  | 128 | 2.3 sec | 1.2 sec 
+| P3.8XLarge |   1  | 256 | 1.2 sec | 0.7 sec
+| P3.8XLarge |   1  | 512 | 0.8 sec | 0.5 sec
+| P3.8XLarge |   1  | 1024 | 0.4 sec | 0.4 sec
+
+##### Memory utilization
```
Comment (roywei) [inline benchmark/sparse/linear_regression/inference_benchmark.md]: Please also remove, this is the CPU memory% when GPU is used for training.

File: benchmark/sparse/linear_regression/training_benchmark.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+| Instance Type | GPUs  | Batch Size  | Keras-MXNet (Time/Epoch) | Keras-TensorFlow (Time/Epoch)  |
+|-----|-----|-----|-----|-----|
+| P3.8XLarge |   1  | 64  | 50.9 sec | 49.9 sec
+| P3.8XLarge |   1  | 128 | 34.8 sec | 33.5 sec 
+| P3.8XLarge |   1  | 256 | 18.8 sec | 23.3 sec
+| P3.8XLarge |   1  | 512 | 12.4 sec | 19.0 sec
+| P3.8XLarge |   1  | 1024 | 9.1 sec | 18.8 sec
+
+##### Memory utilization
+| Instance Type | GPUs  | Batch Size | Keras-MXNet (GPU Utilization %) | Keras-TensorFlow (GPU Utilization %)  |
```
Comment (roywei) [inline benchmark/sparse/linear_regression/training_benchmark.md]: save as above, please remove

File: benchmark/sparse/linear_regression/training_benchmark.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+##### Multi-GPU
+##### Speed
+###### Benchmark results on multi GPU calculated using an average of 5 runs
+| Instance Type | GPUs  | Batch Size | Keras-MXNet (Time/Epoch) | Keras-TensorFlow (Time/Epoch)  |
+|-----|-----|-----|-----|-----|
+| P3.8XLarge |   2  | 512 | 19 sec | Not supported
+| P3.8XLarge |   4  | 1024  | 21.6 sec | Not supported
+
+##### Memory utilization
+###### Reporting total memory utilization by all GPUs
```
Comment (roywei) [inline benchmark/sparse/linear_regression/training_benchmark.md]: same, remove

File: benchmark/sparse/linear_regression/training_benchmark.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+| Instance Type | GPUs  | Batch Size | Keras-MXNet (GPU Utilization %) | Keras-TensorFlow (GPU Utilization %)  |
+|-----|-----|-----|-----|-----|
+| P3.8XLarge |   2  | 512  | 11 | Not supported
```
Comment (roywei) [inline benchmark/sparse/linear_regression/training_benchmark.md]: use per gpu utilization

File: benchmark/sparse/linear_regression/training_benchmark.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+| P3.8XLarge |   4  | 128 | 12 | Not supported
+
+##### Memory consumed
+###### Reporting total memory consumed by all GPUs
+| Instance Type | GPUs  | Batch Size | Keras-MXNet (GPU Memory (MB)) | Keras-TensorFlow (GPU Memory (MB))  |
+|-----|-----|-----|-----|-----|
+| P3.8XLarge |  2  | 512 | 1944.1  | Not supported
```
Comment (roywei) [inline benchmark/sparse/linear_regression/training_benchmark.md]: use per gpu memory

File: benchmark/sparse/linear_regression/training_benchmark.md (hunk 4 above)
Comment (kalyc) [reply benchmark/sparse/linear_regression/training_benchmark.md]: Updated header name to indicate GPU utilization

File: benchmark/sparse/linear_regression/inference_benchmark.md (hunk 3 above)
Comment (kalyc) [reply benchmark/sparse/linear_regression/inference_benchmark.md]: Updated header name to indicate GPU utilization

**Pull request merged**